rcp/org.testeditor.rcp4.bootstrap/src/org/testeditor/rcp4/bootstrap/Activator.java

```diff
@@ -0,0 +1,65 @@
+/*******************************************************************************
+ * Copyright (c) 2012 - 2016 Signal Iduna Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ * 
+ * Contributors:
+ * Signal Iduna Corporation - initial API and implementation
+ * akquinet AG
+ * itemis AG
+ *******************************************************************************/
+package org.testeditor.rcp4.bootstrap;
+
+import java.net.BindException;
+import java.net.ServerSocket;
+
+import org.osgi.framework.BundleActivator;
+import org.osgi.framework.BundleContext;
+
+public class Activator implements BundleActivator {
+
+	private static final String OSGI_HTTP_PORT = "org.osgi.service.http.port";
+	private static BundleContext context;
+
+	static BundleContext getContext() {
+		return context;
+	}
+
+	/*
+	 * (non-Javadoc)
+	 * 
+	 * @see org.osgi.framework.BundleActivator#start(org.osgi.framework.
+	 * BundleContext)
+	 */
+	public void start(BundleContext bundleContext) throws Exception {
+		Activator.context = bundleContext;
+		if (System.getProperty(OSGI_HTTP_PORT) != null) {
```

Comment:
What about the "else" case? I assume that if the property is not set to port should be assigned at all? A small comment below the if would be nice. For example:
```
if (...) {
} // else: if property not set we don't want to assign a port